classes/core_userkey_manager.php

```diff
@@ -122,38 +122,45 @@ public function validate_key($keyvalue) {
             print_error('expiredkey');
         }
 
-        if ($key->iprestriction) {
-            $remoteaddr = getremoteaddr(null);
-
-            if (isset($this->config->ipwhitelist)) {
-                $whitelist = $this->config->ipwhitelist;
-            } else {
-                $whitelist = false;
-            }
-
-            if (empty($remoteaddr) ) {
-                print_error('noip', 'auth_userkey');
-            } else if (!empty($whitelist)) {
-                $ips = explode(';', $whitelist);
-                $whitelisted = false;
-                foreach ($ips as $ip) {
-                    if (address_in_subnet($remoteaddr, $ip)) {
-                        $whitelisted = true;
-                    }
-                }
-                if (!$whitelisted) {
-                    print_error('ipmismatch', 'error', '', null, "Remote address: $remoteaddr\nKey IP: $key->iprestriction");
-                }
-            } else if (!address_in_subnet($remoteaddr, $key->iprestriction)) {
-                print_error('ipmismatch', 'error', '', null, "Remote address: $remoteaddr\nKey IP: $key->iprestriction");
-            }
-        }
-
         if (!$user = $DB->get_record('user', array('id' => $key->userid))) {
             print_error('invaliduserid');
         }
 
+        $this->validate_ip_address($key);
```

Comment:
Why did you move this validation down after getting user from DB?

Reply:
That wasn't intentional, I have changed the order back.